test/fluttertagger_test.dart

```diff
@@ -0,0 +1,272 @@
+              },
+            ),
+          ),
+        ),
+      );
+    });
+
+    testWidgets('initializes without errors', (tester) async {
+      await tester.pumpWidget(testWidget);
+      expect(find.byType(FlutterTagger), findsOneWidget);
+      expect(find.byType(TextField), findsOneWidget);
+    });
+
+    testWidgets('triggers search callback with correct query', (tester) async {
+      String searchQuery = '';
+      String triggerChar = '';
+      onSearch = (query, triggerCharacter) {
+        searchQuery = query;
+        triggerChar = triggerCharacter;
+      };
+
+      testWidget = MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: FlutterTagger(
+              overlay: Container(
+                height: 100,
+                color: Colors.grey,
+                child: const Center(child: Text('Overlay')),
+              ),
+              controller: controller,
+              onSearch: onSearch,
+              builder: (context, key) {
+                return TextField(
+                  key: key,
+                  controller: controller,
+                  onChanged: (value) {},
+                );
+              },
+              triggerCharacterAndStyles: const {
+                '@': TextStyle(color: Colors.blue),
+                '#': TextStyle(color: Colors.green),
+              },
+            ),
+          ),
+        ),
+      );
+
+      await tester.pumpWidget(testWidget);
+
+      final textField = find.byType(TextField);
+      await tester.tap(textField);
+      await tester.pump();
+
+      // Incrementally enter each character to simulate typing
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@t');
+      await tester.pump();
+      await tester.enterText(textField, '@te');
+      await tester.pump();
+      await tester.enterText(textField, '@tes');
+      await tester.pump();
+      await tester.enterText(textField, '@test');
+      await tester.pumpAndSettle();
+
+      expect(searchQuery, 'test');
+      expect(triggerChar, '@');
+    });
+
+    testWidgets('displays overlay when typing a trigger character',
+        (tester) async {
+      await tester.pumpWidget(testWidget);
+
+      final textField = find.byType(TextField);
+      await tester.tap(textField);
+      await tester.pump();
+
+      // Incrementally enter each character to simulate typing
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@t');
+      await tester.pump();
+      await tester.enterText(textField, '@te');
+      await tester.pump();
+      await tester.enterText(textField, '@tes');
+      await tester.pump();
+      await tester.enterText(textField, '@test');
+      await tester.pumpAndSettle();
+
+      expect(find.text('Overlay'), findsOneWidget);
+    });
+
+    testWidgets('formats and displays tagged text correctly', (tester) async {
```

Comment:
Tagged texts usually have a format. For instance, by default, `@6zo22531b866ce0016f9e5tt#Lucky Ebere#` is formatted as `@Lucky Ebere` in the TextField.

I believe this is the kind of behavior this test should verify.

You can also provide a `tagTextFormatter` property to `FlutterTagger` to define custom rules for formatting tags and test against that.

Reply:
Treated.